<launch>
  <arg name="sensorPitch" default="0"/>
  <arg name="vehicleX" default="0"/>
  <arg name="vehicleY" default="0"/>
  <arg name="vehicleZ" default="1.0"/>
  <arg name="vehicleYaw" default="0"/>
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="realtime_factor" default="1.0"/>
  <arg name="gui" default="false"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="verbose" default="false"/>
  <arg name="map_name" default="office"/>
  <arg name="use_gazebo" default="true"/>
  <arg name="sim_name" default="gazebo"/>
  <arg name="sensor_name" default="lidar"/>

  <node pkg="vehicle_simulator" type="vehicleSimulator" name="vehicleSimulator" output="screen">
    <remap from="scan" to="/scan"/>
    <remap from="mavros/local_position/odom" to="/mavros/local_position/odom"/>
    <param name="realtimeFactor" value="$(arg realtime_factor)" />
    <param name="windCoeff" type="double" value="0.05" />
    <param name="maxRollPitchRate" type="double" value="20.0" />
    <param name="rollPitchSmoothRate" type="double" value="0.1" />
    <param name="sensorPitch" value="$(arg sensorPitch)" />
    <param name="vehicleX" value="$(arg vehicleX)" />
    <param name="vehicleY" value="$(arg vehicleY)" />
    <param name="vehicleZ" value="$(arg vehicleZ)" />
    <param name="vehicleYaw" value="$(arg vehicleYaw)" />
    <param name="sim_name" value="$(arg sim_name)" />
    <param name="sensor_name" value="$(arg sensor_name)" />
  </node>

</launch>
